watch/source.go

```diff
@@ -21,19 +21,24 @@
 package watch
 
 import (
+	"errors"
 	"sync"
 
 	"github.com/m3db/m3x/close"
 	"github.com/m3db/m3x/log"
 )
 
+var ErrSourceClosed = errors.New("source closed")
+
 // SourcePollFn provides source data
 type SourcePollFn func() (interface{}, error)
 
 // Source polls data by calling SourcePollFn and notifies its watches on updates
 type Source interface {
 	xclose.SimpleCloser
 
+	// Initialized returns whether the Source was initialized
+	Initialized() bool
```

Comment:
I'm thinking having the caller to figure out a way to wait on it, so you mean return a channel that'll be closed when initializing?